File: divide-a-string-into-groups-of-size-k.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        string temp;
        vector<string>ans;
        while(s.size()>k){
            temp.clear();
            temp=s.substr(0,k);
            s.erase(0,k);
            ans.push_back(temp);
        }
        if(s.size()==k){
            ans.push_back(s);
            return ans;
        }
        while(s.size()!=k){
            s.push_back(fill);
        }
        ans.push_back(s);
        return ans;
    }
};

File: divide-a-string-into-groups-of-size-k.py
class Solution:
    def divideString(self, s: str, k: int, fill: str) -> List[str]:
        
        res=[]
        curr = ""
        for i in range(len(s)):
            if len(curr)==k:
                res.append(curr)
                curr=""
            curr+=s[i]
        if len(curr)!=0:
           res.append(curr+(fill*(k-len(curr))))
        return res